src/robot/model/modelobject.py

```diff
@@ -26,4 +26,33 @@ def __unicode__(self):
     def __repr__(self):
         return repr(str(self))
 
+    def __setstate__(self, state):
+        """Customize the attribute updating when using `copy.copy` or `copy.deepcopy`
+        """
+
+        # We should consider the state format per pickle protocol version
+        # Refer to: https://www.python.org/dev/peps/pep-0307
+        if isinstance(state, tuple) and len(state) == 2:
+            dictstate, slotstate = state
+        else:
+            dictstate = state
+            slotstate = None
```

Comment:
We have `__slots__` and can thus know, based on the referred PEP, that state is two-tuple.